Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py

```diff
@@ -171,23 +171,22 @@ def save_failed_asset_joblogs(workspace: AbstractWorkspace) -> None:
 
 def _check_log_errors_warnings(log_path: str) -> bool:
     """
-    Checks to see if the asset log contains any errors or warnings by reading the last line of the file.
-    Example log line: '~~1643759303647~~1~~00000000000009E0~~AssetBuilder~~S: 0 errors, 1 warnings'
+    Checks to see if the asset log contains any errors or warnings by reading the last line of the file. There are
+    multiple types of log regex. Returns True is no regex is found because something probably went wrong.
+    Example log lines: ~~1643759303647~~1~~00000000000009E0~~AssetBuilder~~S: 0 errors, 1 warnings
+    (Current issue where the expected line isn't always the last line. This function will instead query each line until
+    the issue is resolved).
```

Comment:
Please remove this stray comment or add a reference to an open GHI ticket

This appears to be docstring thrash. Would be better to not claim specifics about how the regex detects the errors in this docstring, and leave the code to track implementation details. In this case, you can also edit the return value to more simply state "True if an error or warning is detected, else False"

Reply:
I don't think a new ticket is necessary, unless this wider regex parsing is **spectacularly** slow.